lib/public-stats.js

```diff
@@ -101,37 +102,108 @@ const updateIndexerQueryStats = async (pgClient, committees) => {
 /**
  * @param {pg.Client} pgClient
  * @param {Iterable<Committee>} committees
+ * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealClients
  */
-const updateDailyDealsStats = async (pgClient, committees) => {
-  let total = 0
-  let indexed = 0
-  let retrievable = 0
+const updateDailyDealsStats = async (pgClient, committees, findDealClients) => {
+  /** @type {Map<string, Map<string, {
+   * tested: number;
+   * majority_found: number;
+   * indexed:  number;
+   * indexed_http: number;
+   * retrievable: number;
+   * }>>} */
+  const minerClientDealStats = new Map()
   for (const c of committees) {
-    total++
+    const { minerId, cid } = c.retrievalTask
+    let clients = findDealClients(minerId, cid)
+    if (!(clients && clients.length)) {
```

Comment:
```suggestion
    if (!clients || !clients.length) {
```

I find this easier to reason about, but don't mind keeping the original version if you prefer it

Reply:
I was half expecting this comment 😄  I am fine either way.

Reply:
This is also happening to me!